src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -548,9 +545,51 @@ private void SetDirty(TimeSpan? maxDelay)
             return applicationData;
         }
 
+        private Dictionary<string, ConfigurationSetting> ReorderData(Dictionary<KeyValueIdentifier, string> cachedInfoLogs = default)
+        {
+            Dictionary<string, ConfigurationSetting> orderedData = new Dictionary<string, ConfigurationSetting>();
+            Dictionary<string, string> logIdentifiers = new Dictionary<string, string>();
+
+            foreach (KeyValuePair<KeyValueIdentifier, ConfigurationSetting> kvp in _mappedData)
+            {
+                orderedData[kvp.Key.Key] = kvp.Value;
+            }
+
+            foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers.Concat(_options.MultiKeyWatchers))
+            {
+                IEnumerable<ConfigurationSetting> currentKeyValues = GetCurrentKeyValueCollection(changeWatcher.Key, changeWatcher.Label, _watchedSettings.Values);
+
+                foreach (ConfigurationSetting setting in currentKeyValues)
+                {
+                    orderedData[setting.Key] = _mappedData[new KeyValueIdentifier(setting.Key, setting.Label)];
+                    logIdentifiers[setting.Key] = setting.Label;
+                }
+            }
+
+            if (cachedInfoLogs != null)
+            {
+                StringBuilder logInfoBuilder = new StringBuilder();
+
+                foreach (KeyValuePair<string, string> kvp in logIdentifiers)
+                {
+                    if (cachedInfoLogs.TryGetValue(new KeyValueIdentifier(kvp.Key, kvp.Value), out string log))
+                    {
+                        logInfoBuilder.AppendLine(log);
+                    }
+                }
+
+                if (logInfoBuilder.Length > 0)
+                {
+                    _logger.LogInformation(logInfoBuilder.ToString().Trim());
```

Comment:
What is this log? Is this for debugging? I don't see any mention of logging in the PR description.

Reply:
It replaces the log that was removed on line 387. I can see how it looks confusing to split these up, but I did it to prevent some issues with passing around params. I could try to move it or add comments to clarify